src/tools/illink/test/Mono.Linker.Tests.Cases/Reflection/MethodUsedViaReflection.cs

```diff
@@ -486,6 +487,37 @@ public static void TestNameWithIntAndTypeAndModifiers()
             }
         }
 
+        // GetMethod(string name, int genericParameterCount, BindingFlags bindingAttr, Type[] types)
+        [Kept]
+        class GetMethod_Name_GenericParameterCount_BindingAttr_Types
+        {
+            private static int OnlyCalledViaReflection()
+            {
+                return 42;
+            }
+            private int OnlyCalledViaReflection(int foo)
+            {
+                return 43;
+            }
+            [Kept]
+            public int OnlyCalledViaReflection(int foo, int bar)
+            {
+                return 44;
+            }
+            [Kept]
+            public static int OnlyCalledViaReflection(int foo, int bar, int baz)
+            {
+                return 45;
+            }
+
+            [Kept]
+            public static void TestNameWithIntAndBindingFlagsAndTypes()
+            {
+                var method = typeof(GetMethod_Name_GenericParameterCount_BindingAttr_Types).GetMethod("OnlyCalledViaReflection", 0, BindingFlags.Public | BindingFlags.Static, new Type[] { typeof(int), typeof(int), typeof(int) });
```

Comment:
The test seems wrong; the `OnlyCalledViaReflection` methods must be generic because the integer and `Type[]` parameters in `Type.GetMethod` refer to generic parameters.